physics/MP/GFDL/v3_2022/gfdl_cloud_microphys_v3_mod.F90

```diff
@@ -0,0 +1,7377 @@
+        procedure ies_t
+        procedure iqs_trho
+        procedure iqs_ptqv
+    end interface iqs
+
+    interface mhc
+        procedure mhc3
+        procedure mhc4
+        procedure mhc6
+    end interface mhc
+
+    interface wet_bulb
+        procedure wet_bulb_dry
+        procedure wet_bulb_moist
+    end interface wet_bulb
+
+    ! -----------------------------------------------------------------------
+    ! public subroutines, functions, and variables
+    ! -----------------------------------------------------------------------
+
+    public :: gfdl_cloud_microphys_v3_mod_init
+    public :: gfdl_cloud_microphys_v3_mod_driver
+    public :: gfdl_cloud_microphys_v3_mod_end
+    public :: cld_sat_adj, cld_eff_rad, rad_ref
+    public :: qs_init, wqs, mqs, mqs3d
+    public :: c_liq, c_ice, rhow, wet_bulb
+    public :: cv_air, cv_vap, mtetw
+    public :: hlv, hlf, tice
+
+    ! -----------------------------------------------------------------------
+    ! precision definition
+    ! -----------------------------------------------------------------------
+
+    integer, parameter :: r8 = 8 ! double precision
+
+    ! -----------------------------------------------------------------------
+    ! initialization conditions
+    ! -----------------------------------------------------------------------
+
+    logical :: tables_are_initialized = .false. ! initialize satuation tables
+
+    ! -----------------------------------------------------------------------
+    ! physics constants
+    ! -----------------------------------------------------------------------
+
+    real, parameter :: grav = 9.80665 ! acceleration due to gravity (m/s^2), ref: IFS
+
+    real, parameter :: rgrav = 1.0 / grav ! inversion of gravity acceleration (s^2/m)
+
+    real, parameter :: pi = 4.0 * atan (1.0) ! ratio of circle circumference to diameter
+
+    real, parameter :: boltzmann = 1.38064852e-23 ! boltzmann constant (J/K)
+    real, parameter :: avogadro = 6.02214076e23 ! avogadro number (1/mol)
+    real, parameter :: runiver = avogadro * boltzmann ! 8.314459727525675, universal gas constant (J/K/mol)
+    real, parameter :: mmd = 2.89644e-2 ! dry air molar mass (kg/mol), ref: IFS
+    real, parameter :: mmv = 1.80153e-2 ! water vapor molar mass (kg/mol), ref: IFS
+
+    real, parameter :: rdgas = 287.05 ! gas constant for dry air (J/kg/K): ref: GFDL, GFS
+    real, parameter :: rvgas = 461.50 ! gas constant for water vapor (J/kg/K): ref: GFDL, GFS
+    !real, parameter :: rdgas = runiver / mmd ! 287.0578961596192, gas constant for dry air (J/kg/K)
+    !real, parameter :: rvgas = runiver / mmv ! 461.52213549181386, gas constant for water vapor (J/kg/K)
+
+    real, parameter :: zvir = rvgas / rdgas - 1. ! 0.6077667316114637
+    real, parameter :: eps = rdgas / rvgas ! 0.6219934994582882
+    real, parameter :: epsm1 = rdgas / rvgas - 1. ! -0.3780065005417118
+
+    real, parameter :: tice = 273.15 ! freezing temperature (K): ref: GFDL, GFS
```

Comment:
This is double redefining. It was redefined in the GFDL MP specific module then redefined here. Why?

Reply:
Fixed.